### PR TITLE
Allow overwriting possible nlog configuration file paths

### PR DESCRIPTION
**Pull request by 304NotModified**
- [x] `GetDefaultCandidateConfigFileNames` to public

### CODE REVIEW

**Comment by 304NotModified** [comment]
> Instead of providing with set and get property. We can make this variable private and Can create a method "OverwriteConfigFileNames" and set the private variable internally. So it is very obvious API Users. 

Nice idea, but what is the benefit over a get and setter?

**Comment by Page-Not-Found** [comment]
Based on coding standard (https://msdn.microsoft.com/en-us/library/ms182327.aspx), collections should not have setter. This is one reason for the suggestion.

**Comment by 304NotModified** [comment]
Good point :)

But I could be a getter only indeed

**Comment by 304NotModified** [comment]
@Page-Not-Found your opinion please :)

File: src/NLog/LogFactory.cs [hunk 1]
```diff
@@ -865,10 +872,51 @@ protected virtual void Dispose(bool disposing)
...
+        /// <param name="filePaths">The filepaths to the possible config file</param>
+        public void SetCandidateConfigFilePaths(IEnumerable<string> filePaths)
+        {
+            candidateConfigFilePaths = new List<string>();
+
+            if (filePaths != null)
+            {
+                foreach (var filePath in filePaths)
+                {
+                    candidateConfigFilePaths.Add(filePath);
```
Comment (Page-Not-Found) [inline src/NLog/LogFactory.cs]: You could just use "AddRange" instead of loop and a add.

File: src/NLog/LogFactory.cs [hunk 2]
```diff
@@ -865,10 +872,51 @@ protected virtual void Dispose(bool disposing)
...
+                }
+            }
+        }
+        /// <summary>
+        /// Clear the candidate file paths and return to the defaults.
+        /// </summary>
+        public void ClearCandidateConfigFilePath()
+        {
```
Comment (Page-Not-Found) [inline src/NLog/LogFactory.cs]: Can be renamed "Reset...." so it is matching with the "Set...." method.

File: src/NLog/LogManager.cs [hunk 3]
```diff
@@ -178,6 +178,32 @@ public static GetCultureInfo DefaultCultureInfo
...
+            return factory.GetCandidateConfigFilePaths();
+        }
+
+        /// <summary>
+        /// Overwrite the paths (including filename) for the possible NLog config files.
+        /// </summary>
+        /// <param name="filePaths">The filepaths to the possible config file</param>
+        public static void SetCandidateConfigFilePaths(IEnumerable<string> filePaths)
+        {
+            factory.SetCandidateConfigFilePaths(filePaths);
```
Comment (Page-Not-Found) [inline src/NLog/LogManager.cs]: Same as above.

**Comment by Page-Not-Found** [comment]
@304NotModified  Minor suggestions.

**Comment by 304NotModified** [comment]
Thanks @Page-Not-Found !

One thing, the name of the method. `LogManager.GetCandidateConfigFilePaths`, it really visible (next to autocomplete of `GetCurrentClassLogger` : 

image

Maybe it should move it to somewhere else? What do you think?

**Comment by Page-Not-Found** [comment]
I am not sure. if it is public right now. If it is something added for this Issue. I would recommend hiding it by marking as internal.

File: src/NLog/LogFactory.cs (hunk 1 above)
Comment (304NotModified) [inline src/NLog/LogFactory.cs]: Thanks, it was first a `IList` but with the move to `List`, AddRange is indeed better

File: src/NLog/LogFactory.cs (hunk 2 above)
Comment (304NotModified) [inline src/NLog/LogFactory.cs]: That name is more clear indeed, thx1

Comment (Page-Not-Found) [inline src/NLog/LogFactory.cs]: I agree. You can ignore the comment and go ahead.

**Comment by 304NotModified** [comment]
It should be public, as I need it for NLog.Extensions.Logging

But I think its better to move the LogManager methods to `XmlLoggingConfiguration` or `LoggingConfiguration`. What do you think?

**Comment by 304NotModified** [comment]
moved it to XMLLoggingConfiguration, it's easier to find and less pollutes LogManager IMO

**Pull request merged**